cmd/uplink/cmd/cp.go

```diff
@@ -49,7 +49,12 @@ func upload(ctx context.Context, bs buckets.Store, srcFile string, destObj *url.
 	destObj.Path = cleanAbsPath(destObj.Path)
 	// if object name not specified, default to filename
 	if strings.HasSuffix(destObj.Path, "/") {
-		destObj.Path = path.Join(destObj.Path, path.Base(srcFile))
+		// Check for absolute filepath on Windows
+		if filepath.IsAbs(srcFile) {
+			destObj.Path = path.Join(destObj.Path, filepath.Base(srcFile))
+		} else {
+			destObj.Path = path.Join(destObj.Path, path.Base(srcFile))
+		}
```

Comment:
I think we might need some proper `type` for handling the variability in the Path behavior. I suspect other places start to hit similar issues.

Rough sketch for the path type:

```
type Path struct {
    Local     bool // whether it represents a local system or remote path
    Delimiter byte // usually '/', for windows '\'
    Schema    string // the schema for non-local
    Path      string
}

func (path Path) Join(segments ...Path) Path { ... }
func (path Path) Folder() Path { ... }
func (path Path) Base() Path { ... }
```

Reply:
Maybe we should take a look into the AWS CLI Source code to see how they handled it?